addon/ng2/utilities/ast-utils.ts

```diff
@@ -0,0 +1,54 @@
+import * as ts from 'typescript';
+import { InsertChange } from './change';
+
+/**
+* Find all nodes from the AST in the subtree of node of SyntaxKind kind.
+* @param node 
+* @param kind (a valid index of ts.SyntaxKind enum, eg ts.SyntaxKind.ImportDeclaration)
+* @return all nodes of kind kind, or [] if none is found
+*/
+export function findNodes (node: ts.Node, kind: number, arr: ts.Node[] = []): ts.Node[] {
+  if (node) {
+    if (node.kind === kind) {
+      arr.push(node);
+    }
+    node.getChildren().forEach(child => findNodes(child, kind, arr));
+  }
+  return arr;
+}
+
+/**
+ * @param nodes (nodes to sort)
+ * @return (nodes sorted by their position from the source file 
+ *          or [] if nodes is empty)
+ */
+export function sortNodesByPosition(nodes: ts.Node[]): ts.Node[] {
```

Comment:
Maybe instead of having a function which sorts the array, have a function which can be passed to `sort()`. That way someone can do:

`myArray.sort(nodesByPosition)`.

This I think is a lot clearer, since your function can be pure instead of mutating its arguments (never a good idea).